src/ros_filter.cpp

```diff
@@ -2971,12 +2973,32 @@ namespace RobotLocalization
 
     // Walk back through the queue until we reach a filter state whose time stamp is less than or equal to the
     // requested time. Since every saved state after that time will be overwritten/corrected, we can pop from
-    // the queue.
+    // the queue. If the history is insufficiently short, we just take the oldest state we have.
+    FilterStatePtr lastHistoryState;
     while (!filterStateHistory_.empty() && filterStateHistory_.back()->lastMeasurementTime_ > time)
```

Comment:
This is all working, so I'm not suggesting you change anything. But would it have been simpler just to change `while` loop condition to `(filterStateHistory_.size() > 1) && (filterStateHistory_.back()->lastMeasurementTime_ > time)` to prevent clearing the entire `filterStateHistory_` container?

Reply:
So the premise is that, if the measurement we get is older than the oldest state in our history, we want to replace the oldest state in our history. If the history is too short, we will clear it, and then when we try to fuse the old measurement that prompted the reversion, we will do an immediate correction at the time in question, and that will become our new state for that time, and it will get stored in the history. If I don't clear the history when its length is insufficient (i.e., if we keep the one at the back), we'll have two states with the same time stamp after we process the "old" measurement.

Note that this really only will affect us within the first 0.4 seconds of execution (current history length), after which time we will always have sufficient history.